testprojects/mailhog/src/test/kotlin/io/cloudflight/testresources/springboot/mailhog/ApplicationTest.kt

```diff
@@ -0,0 +1,48 @@
+package io.cloudflight.testresources.springboot.mailhog
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.http.converter.KotlinSerializationStringHttpMessageConverter
+import org.springframework.mail.MailSender
+import org.springframework.mail.SimpleMailMessage
+import org.springframework.web.client.RestTemplate
+
+@SpringBootTest
+class ApplicationTest(
+    @Autowired private val mailSender: MailSender,
+    @Value("\${test-resources.mailhog.api-url}") private val mailhogApiUrl: String,
+) {
+    @Serializable
+    data class MailhogResponse(val count: Int)
+
+    val restTemplate = RestTemplate(listOf(object : KotlinSerializationStringHttpMessageConverter<Json>(Json {
```

Comment:
I'd probably put that RestTemplate (encapsulated into a bean interface) into the springboot-testresources-mailhog project so that this can be reused by clients that want to fetch for mails.

Reply:
but probably it would be cleaner in the end to publish a separate kotlin-wrapper-library for the mailhog API, and not polluting this library with that one.

Reply:
I'd opt to create a libary in a seperate project and keep this code in here for now.